### PR TITLE
Preselect boundary based on the user data source

### LINKED ISSUE #12623: Make DO multiple measure auto-select best geometry
DO has a quite interesting system to automatically detect the best geometry for a given dataset and augmentation. Although Builder has this information, it is not currently doing anything with it. It would be great if we could make the geometry that DO considers the best to be the default in the UI.

When asking for the list of geometries in a multiple measure analysis, DO returns something like

```json
[
        {
            "geom_id": "us.census.tiger.county_clipped",
            "geom_name": "Shoreline clipped US County",
            "geom_weight": "5.010000000000000000208166817",
            "score": 48.9005821201008,
            [...]
        },
        {
            "geom_id": "us.census.tiger.puma_clipped",
            "geom_name": "Shoreline clipped US Census Public Use Microdata Areas",
            "geom_weight": "4.010000000000000000208166817",
            "score": 40.4863299235498,
            [...]
        },
        {
            "geom_id": "us.census.tiger.congressional_district_clipped",
            "geom_name": "Shoreline clipped US Congressional Districts",
            "geom_weight": "2.010000000000000000208166817",
            "score": 30.973792235478125,
            [...]
        },
        {
            "geom_id": "us.census.tiger.state_clipped",
            "geom_name": "Shoreline clipped US States",
            "geom_weight": "1.010000000000000000208166817",
            "score": 26.004888297362797,
            [...]
        }
    ]
```

Builder uses the `weight` field to order the geometries in the slider component. `score` is the result of the auto-detection algorithm, the highest score, the best. So, in this example, the county-level geometry should be selected as a default.

cc @noguerol

### PR DESCRIPTION
**Pull request by ethervoid**
Closes #12623 

Now we use the information provided by the getavailablegeometries
to pre-select the best boundary for the provided user data

### CODE REVIEW

File: lib/assets/core/javascripts/cartodb3/data/data-observatory/boundaries-collection.js [hunk 1]
```diff
@@ -32,6 +35,17 @@ module.exports = BaseCollection.extend({
...
   },
 
+  getPreSelectedBoundaryValue: function () {
+    var boundary = this.models.filter(
+      function (boundary) { return boundary.attributes.rank === 1; }
+    ).shift();
+    if (boundary !== undefined) {
+      return boundary.attributes.val;
+    } else {
+      return undefined;
```
Comment (nobuti) [inline lib/assets/core/javascripts/cartodb3/data/data-observatory/boundaries-collection.js]: This is implicit if no return is present.

Comment (nobuti) [reply lib/assets/core/javascripts/cartodb3/data/data-observatory/boundaries-collection.js]: We can simplify this as:
```
return boundary && boundary.attributes && boundary.attributes.val;
```

File: lib/assets/core/javascripts/cartodb3/components/form-components/editors/slider/slider.js [hunk 2]
```diff
@@ -44,6 +44,10 @@ Backbone.Form.editors.Slider = Backbone.Form.editors.Base.extend({
       throw new Error('values and labels should have the same length');
     }
 
+    if (this.options.value !== undefined) {
```
Comment (nobuti) [inline lib/assets/core/javascripts/cartodb3/components/form-components/editors/slider/slider.js]: Can we add a test for this?

**Comment by nobuti** [comment]
Applied the comments, let me know if you see something odd.

**Comment by ethervoid** [comment]
Wow thank you man, I was going to make that changes. I didn't add tests nor NEWS to make a first check if my solution was right :)

**Comment by javitonino** [comment]
Mostly works ok, but I found some acceptance scenarios that are doing weird stuff:

1. Add two points very close together (< 1km) in US
2. Enrich from DO -> Total population. Timespan is autoselected to 2015-2015, and the smallest boundary for that is county.
3. Switch to 2011-2015. County is selected. It should have selected block group. It is keeping the last selection.

Do you feel we can improve this behaviour? Something like reselecting the default if the set of available boundaries changes or something along those lines?

**Comment by ethervoid** [comment]
Hum, if you didn't save/launch the analysis and store the value we it should be recalculated unless the change of timespan doesn't trigger the geometries part. Worth a check, thank you :)

**Comment by ethervoid** [comment]
I've changed this a bit check the commit [this commit](https://github.com/CartoDB/cartodb/pull/12948/commits/9462ee71260b7dfb1b90baeda67d8638a5962711) message for review

// @javitonino @nobuti

**Comment by nobuti** [comment]
Looks great!

**Comment by ethervoid** [comment]
I've tested it in staging and looks good :)

**Comment by javitonino** [comment]
Splendid!

**Pull request merged**